### PR TITLE
Various Solaris helpers

### PR DESCRIPTION
**Pull request by schisamo**
Pretty self-explanatory. Please hold off on merging until @scotthain and @curiositycasualty have given a :+1:.

### CODE REVIEW

**Comment by curiositycasualty** [comment]
LGTM :+1:

File: lib/chef/sugar/architecture.rb [hunk 1]
```diff
@@ -40,6 +40,26 @@ def _32_bit?(node)
...
+          .include?(node['kernel']['machine'])
+      end
+
+      #
+      # Determine if the current architecture is SPARC.
+      #
+      # @return [Boolean]
+      #
+      def sparc?(node)
+        %w(sun4u)
```
Comment (scotthain) [inline lib/chef/sugar/architecture.rb]: This may also return sun4v on certain SPARC machines

**Comment by scotthain** [comment]
Other than the sun4u/sun4v thing - looks great! We just need to add the possibility of sun4v.

File: lib/chef/sugar/architecture.rb (hunk 1 above)
Comment (curiositycasualty) [inline lib/chef/sugar/architecture.rb]: Nice catch

**Comment by scotthain** [comment]
:+1:

**Comment by schisamo** [comment]
@sethvargo This should be ready to :shipit: when you approve.

**Comment by sethvargo** [comment]
Woah. I go on vacation for a week and everyone has new pictures!? :smile:

**Pull request merged**